src/CommunityPatch/Patches/Feats/AseraiCheapCaravansPatch.cs

```diff
@@ -0,0 +1,87 @@
+        private static readonly MethodInfo PrefixPatchMethodInfo = AccessTools.Method(typeof(AseraiCheapCaravansPatch), nameof(Prefix));
+        private static readonly MethodInfo PostfixPatchMethodInfo = AccessTools.Method(typeof(AseraiCheapCaravansPatch), nameof(Postfix));
+
+        private static readonly byte[][] TargetHashes = {
+            new byte[] {
+                // e1.2.0.226271
+                0x72, 0x71, 0xCF, 0x96, 0x32, 0xA5, 0x98, 0x86,
+                0xAB, 0x61, 0xDD, 0x80, 0x8E, 0x36, 0xCC, 0xF2,
+                0x77, 0x37, 0x51, 0xC3, 0x5B, 0x6A, 0x46, 0xF6,
+                0x22, 0x1B, 0x88, 0xED, 0x36, 0xFB, 0xAA, 0x7A
+            }
+        };
+
+        public override IEnumerable<MethodBase> GetMethodsChecked() {
+            yield return AseraiCheapCaravansPatch.TargetMethodInfo;
+        }
+        
+        public override bool? IsApplicable(Game game) {
+            // This patch does not modify the target function with Prefix, which only provides state data for Postfix
+            // It should be compatible with other patches
+
+            var hash = TargetMethodInfo.MakeCilSignatureSha256();
+            return hash.MatchesAnySha256(TargetHashes);
+        }
+        
+        public override void Apply(Game game) {
+            if (Applied) return;
+
+            CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo,
+                prefix: new HarmonyMethod(PrefixPatchMethodInfo),
+                postfix: new HarmonyMethod(PostfixPatchMethodInfo));
+                
+                Applied = true;
+        }
+        
+        public override void Reset() { }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void Prefix(out int __state) {
+            // get player gold and make it available for Postfix
+            __state = Hero.MainHero.Gold;
+        }
+
+        static void Postfix(int __state) {
+            // presume the player successfully purchased a caravan if their gold is less than it was at the beginning
+            // of the target function call
+            if (Hero.MainHero.Gold < __state &&
+                Hero.MainHero.IsHumanPlayerCharacter &&
+                Hero.MainHero.CharacterObject.GetFeatValue(DefaultFeats.Cultural.AseraiCheapCaravans)) {
+                
+                int creditAmount = 0;
+                if (DefaultFeats.Cultural.AseraiCheapCaravans.EffectBonus == 0f) {
+                    // As of e1.2.0.226271, Aserai bonus is not set
+                    // Assume 30% caravan discount per character creation screen text
+                    creditAmount = (int) (0.30f * (__state - Hero.MainHero.Gold));
+                }
+                else {
+                    // If bonus is set, use it
+                    creditAmount = (int) ((1.0f - DefaultFeats.Cultural.AseraiCheapCaravans.EffectBonus) * (__state - Hero.MainHero.Gold));
```

Comment:
If they did set it to 30% someday, wouldn't this result in you getting a 1.0 - 0.3 = 70% discount instead of 30%? Unless they changed it in the beta, the other numbers in the feats file seem to be the modifier itself after all.

_TaleWorlds.CampaignSystem.DefaultFeats, TaleWorlds.CampaignSystem.dll, game e1.1.2_:

```
      ..., "{=ByrkbDoK}10% extra speed bonus for horsemen on campaign map.", 0.1f, ...
      ..., "{=Impk9uLu}20% less speed penalty from snow.", 0.2f, ...
```

```suggestion
                    creditAmount = (int) (DefaultFeats.Cultural.AseraiCheapCaravans.EffectBonus * (__state - Hero.MainHero.Gold));
```

Reply:
As a factor, it makes sense to be .7 ... The IsApplicable check could verify the EffectBonus is .7

Reply:
@iPherian , this is a bug in my code.  Will fix

Reply:
Admittedly i'm not absolutely certain. Here is the entire feat code from the base game. Happy to let others interpret what it means:

```
private void InitializeAll()
    {
      this._cultureBattanianForestAgility.Initialize("{=g4jzOBYX}Battanian Forest Agility", "{=5axKTUpX}Forests give 10% less speed penalty to parties", 0.1f, FeatObject.AdditionType.AddFactor);
      this._cultureVlandianBattlePerception.Initialize("{=2pzjOTtj}Vlandian Battle Perception", "{=bL5YPUbd}20% more upgrade XP to troops from battles.", 0.2f, FeatObject.AdditionType.AddFactor);
      this._cultureKhuzaitCavalryAgility.Initialize("{=egNhD6fF}Khuzait Cavalry Agility", "{=ByrkbDoK}10% extra speed bonus for horsemen on campaign map.", 0.1f, FeatObject.AdditionType.AddFactor);
      this._cultureSturgianSnowAgility.Initialize("{=lMIi4CXO}Sturgian Snow Agility", "{=Impk9uLu}20% less speed penalty from snow.", 0.2f, FeatObject.AdditionType.AddFactor);
      this._cultureAseraiCheapCaravans.Initialize("{=NJmTcbOI}Aserai Cheap Caravans", "{=7kGGgkro}Caravans are 30% cheaper to build. 10% less trade penalty.", 0.0f, FeatObject.AdditionType.AddFactor);
      this._cultureEmpireConstructionExpert.Initialize("{=b7W48wVD}Empire Construction Expert", "{=GIljHGh9}20% construction speed bonus to town projects, wall repairs and siege engines.", 0.2f, FeatObject.AdditionType.AddFactor);
    }
```